### DISCUSSION TITLE
How to upgrade one of the OSes installed by PINN?

### DISCUSSION THREAD

**Discussion by glorifyday** [procount/pinn#704] [Q&A] [answered]
I have a PINN multiboot installation containing several OSes. It was created using the PINN resize tool on a 64 GB micro SD card.

One of the installed OSes is KonstaKANG Linage OS (Android). Version 18, I believe.

Since the installation, a new version (20) of Linage OS was published.
How to cleanly replace the old version with the new one, using the same partition(s), used by the old one?

I don't know if this is relevant to my question, but since the PINN installation I haven't booted the Linage OS yet at all.

**Comment by procount** [comment]
I haven't converted LineageOS 20 to PINN format yet, but when it has been done, you should just be able to replace v18 with v20 (assuming it has the same partition layout and the new partition sizes fit within the allocations you gave it).

**Comment by glorifyday** [comment]
Thanks for your answer. Are you planning to convert LineageOS 20 to PINN in a near future?

What about LineageOS 19? (edit: I checked, it is available on the PINN resize webpage)

Using PINN resize tool I provided the LinageOS with some extra space, so the only problem would emerge, if new OS version required a different number of partitions.

Is there any tutorial how to replace one OS with another using PINN? Can this be done without reinstalling ALL installed systems (this was a problem with NOOBS, I believe)?

**Comment by procount** [comment, marked as answer]
The PINN User Guide can be found at https://github.com/procount/pinn/blob/master/README_PINN.md
You particularly want to look at: https://github.com/procount/pinn/blob/master/README_PINN.md#replace-individual-oses

**Comment by glorifyday** [comment]
Thank you!
